packages/driver/src/cy/commands/request.js

```diff
@@ -277,7 +277,23 @@ module.exports = (Commands, Cypress, cy, state, config) => {
       // because we're handling timeouts ourselves
       cy.clearTimeout('http:request')
 
-      return Cypress.backend('http:request', requestOpts)
+      return Promise.try(() => {
+        // https://github.com/cypress-io/cypress/issues/6178
+        // Check if body is Blob.
+        // construct.name is added because the parent of the Blob is not the same Blob
+        // if it's generated from the test spec code.
+        if (requestOpts.body instanceof Blob || requestOpts?.body?.constructor.name === 'Blob') {
+          return Cypress.Blob.blobToBase64String(requestOpts.body).then((str) => {
```

Comment:
Do you want to rewrite this so that `cy.request` is an async/await function? We are trying to modernize the codebase's Promise usage, and this looks like a prime opportunity.

Reply:
I checked the code and concluded that it cannot be done in the current situation because of [`Bluebird.timeout()`](http://bluebirdjs.com/docs/api/timeout.html). Timeout cannot be handled with async/await because they wait forever for promise resolving/rejecting. 

To refactor, we might need to create a wrapper function that handles timeout.

Example:

```js
const wrapper = async (options) => {
  return Promise.race([
    new Promise(), // handles command logic
    new Promise(), // handles timeout. Throws `TimeoutError`.
  ])
}
```